core/src/main/java/io/questdb/griffin/SqlOptimiser.java

```diff
@@ -5198,15 +5208,11 @@ else if (!(isEffectivelyConstantExpression(node) && n > 1)) {
                 if (qc.getAst().type != LITERAL && qc.getAst().type != FUNCTION && qc.getAst().type != OPERATION) {
                     columnsAndFunctionsOnly = false;
                 }
-                if (qc.getAst().type == FUNCTION
-                        && isHourKeyword(qc.getAst().token) && qc.getAst().paramCount == 1 && qc.getAst().rhs.type == LITERAL) {
-                    hourFunctionKeyCount++;
-                }
                 if (qc.getAst().type == FUNCTION || qc.getAst().type == OPERATION) {
                     totalFunctionKeyCount++;
                 }
             }
-            boolean singleHourFunctionKey = totalFunctionKeyCount == 1 && hourFunctionKeyCount == 1;
+            boolean singleHourFunctionKey = totalFunctionKeyCount == 1;
```

Comment:
This variable no longer makes sense. Previously we checked for single hour() function key and, if that was the case, we tried to merge the inner model into the group by model. With this change, we avoid the merge in case of any single function key, not only hour(). Is that on purpose?

Reply:
this is a good spot, the check can be simplified significantly.

Reply:
done in https://github.com/questdb/questdb/pull/4690/commits/0bc1174ffe21cb7f4f5b2a7b7b1ea10a6e7e97b1